client/src/redux/reducers/viewOptionsReducer.js

```diff
@@ -0,0 +1,211 @@
+        sortField: action.sortType.id,
+        sortAscending: true
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewToolbarTypes.TOGGLE_SORT_ASCENDING:
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        sortAscending: !state[action.viewType].sortAscending
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.VIEW_FIRST_PAGE:
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        currentPage: 1
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.VIEW_LAST_PAGE:
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        currentPage: state[action.viewType].totalPages
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.VIEW_PREVIOUS_PAGE:
+      if (state[action.viewType].currentPage < 2) {
+        return state;
+      }
+
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        currentPage: state[action.viewType].currentPage - 1
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.VIEW_NEXT_PAGE:
+      if (
+        state[action.viewType].currentPage >= state[action.viewType].totalPages
+      ) {
+        return state;
+      }
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        currentPage: state[action.viewType].currentPage + 1
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.VIEW_PAGE_NUMBER:
+      if (
+        !Number.isInteger(action.pageNumber) ||
+        action.pageNumber < 1 ||
+        action.pageNumber > state[action.viewType].totalPages
+      ) {
+        return state;
+      }
+
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        currentPage: action.pageNumber
+      });
+      return Object.assign({}, state, updateState);
+
+    case viewPaginationTypes.SET_PER_PAGE:
+      updateState[action.viewType] = Object.assign({}, state[action.viewType], {
+        pageSize: action.pageSize
+      });
+      return Object.assign({}, state, updateState);
+
+    case credentialsTypes.GET_CREDENTIAL_FULFILLED:
+    case credentialsTypes.GET_CREDENTIALS_FULFILLED:
+      updatePageCounts(viewTypes.CREDENTIALS_VIEW, action.payload.count);
+      return Object.assign({}, state, updateState);
+
+    case sourcesTypes.GET_SOURCE_FULFILLED:
+    case sourcesTypes.GET_SOURCES_FULFILLED:
+      updatePageCounts(viewTypes.SOURCES_VIEW, action.payload.count);
+      return Object.assign({}, state, updateState);
+
+    case scansTypes.GET_SCAN_FULFILLED:
+    case scansTypes.GET_SCANS_FULFILLED:
+      updatePageCounts(viewTypes.SCANS_VIEW, action.payload.count);
+      return Object.assign({}, state, updateState);
+
+    default:
+      return state;
+  }
+}
```

Comment:
Note: 
Nice work on this! 

What's interesting is that now looking at it running, it relates to some of the reducer syntax for the views we came up with, but approached from a parallel perspective. I think we could eventually end up merging these two parallel concepts.